src/react-components/misc/LevelBar.js

```diff
@@ -1,24 +1,32 @@
-import React from "react";
+import React, { useRef, useEffect } from "react";
 import PropTypes from "prop-types";
+import classNames from "classnames";
 import styles from "./LevelBar.scss";
 
-export function LevelBar({ width, height, level }) {
+export function LevelBar({ className, level }) {
+  const ref = useRef();
+
+  useEffect(
+    () => {
+      const pct = level * 100;
+      if (ref.current.clientWidth > ref.current.clientHeight) {
+        ref.current.style.clipPath = `polygon(0% 100%, ${pct}% 100%, ${pct}% 0%, 0% 0%)`;
+      } else {
+        ref.current.style.clipPath = `polygon(0% 100%, 100% 100%, 100% ${100 - pct}%, 0% ${100 - pct}%)`;
+      }
+    },
+    [ref, level]
```

Comment:
Ah reading the rest of the code its obvious why this didn't matter, level is constantly change and this component is being re-rendered every 50ms anyway...

Reply:
Yep, probably this would never be an issue in this specific case as we are constantly updating but it's good to know and get used to it. I've updated to use a callback ref.